src/main/java/org/folio/circulation/domain/LoanService.java

```diff
@@ -0,0 +1,51 @@
+package org.folio.circulation.domain;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.policy.LoanPolicy;
+import org.folio.circulation.domain.policy.library.ClosedLibraryStrategyService;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.support.Result.succeeded;
+
+public class LoanService {
+
+  private final ClosedLibraryStrategyService closedLibraryStrategyService;
+
+  public LoanService(Clients clients) {
+    closedLibraryStrategyService = ClosedLibraryStrategyService.using(clients,
+      DateTime.now(DateTimeZone.UTC), false);
+  }
+
+  public CompletableFuture<Result<LoanAndRelatedRecords>> handleRecall(
+    LoanAndRelatedRecords records) {
+    RequestQueue requestQueue = records.getRequestQueue();
+    Collection<Request> requests = requestQueue.getRequests();
+
+    if(requests.isEmpty()) {
+      return completedFuture(succeeded(records));
+    }
+    /*
```

Comment:
Is this comment still correct / accurate given the code has now been moved?

Reply:
The comment is still correct, this code from LoanRepository and the only thing that was add is apply Closed Library Due Date Management for recall request. This is where this bug was, missing part with closed library due date management